pkg/exporter/syslogexporter/exporter.go

```diff
@@ -102,32 +106,71 @@ func (se *syslogexporter) getTimestamp(record plog.LogRecord) time.Time {
 }
 
 func (se *syslogexporter) pushLogsData(ctx context.Context, ld plog.Logs) error {
-	s, err := Connect(se.logger, se.config, se.tlsConfig, se.hostname, se.pid, se.app)
-	if err != nil {
-		return fmt.Errorf("error connecting to syslog server: %s", err)
+	type droppedResourceRecords struct {
+		resource pcommon.Resource
+		records  []plog.LogRecord
 	}
-	defer s.Close()
+	var (
+		errs    []error
+		dropped []droppedResourceRecords
+	)
 	rls := ld.ResourceLogs()
 	for i := 0; i < rls.Len(); i++ {
 		rl := rls.At(i)
-		slgs := rl.ScopeLogs()
-		for i := 0; i < slgs.Len(); i++ {
-			slg := slgs.At(i)
-			for j := 0; j < slg.LogRecords().Len(); j++ {
-				lr := slg.LogRecords().At(j)
-				formattedLine := se.logsToMap(lr)
-				timestamp := se.getTimestamp(lr)
-				err = s.Write(formattedLine, timestamp)
-				if err != nil {
-					//TODO: add handling of failures as it is in sumologic exporter
-					return err
-				}
+		if droppedRecords, err := se.sendSyslogs(ctx, rl); err != nil {
+			dropped = append(dropped, droppedResourceRecords{
+				resource: rl.Resource(),
+				records:  droppedRecords,
+			})
+			errs = append(errs, err)
+		}
+	}
+	if len(dropped) > 0 {
+		ld = plog.NewLogs()
+		for i := range dropped {
+			rls := ld.ResourceLogs().AppendEmpty()
+			logRecords := rls.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+			dropped[i].resource.MoveTo(rls.Resource())
+			for j := 0; j < len(dropped[i].records); j++ {
+				dropped[i].records[j].MoveTo(logRecords)
```

Comment:
Please add unit tests for this

Reply:
do we need a test syslog server in our tests (mock) to write tests for the `pushLogsData`?

Reply:
will add the test for this in a new PR